src/db.go

```diff
@@ -124,6 +125,72 @@ func (db *dbManager) fetchPositions(id int) ([]position, error) {
 	return positions, nil
 }
 
+// fetchSample returns evenly distributed sample positions, excluding the most recent 100 positions.
+// The positions are ordered by id in ascending order.
+func (db *dbManager) fetchSample(count int) ([]position, error) {
+	// First, get the stats we need
+	const statsQuery = /* sql */ `
+		WITH excluded_positions AS (
+			SELECT id FROM positions ORDER BY id DESC LIMIT 100
+		)
+		SELECT 
+			MIN(id) as first_id, --first id
+			COUNT(*) as total_count --total count
+		FROM positions
+		WHERE id NOT IN (SELECT id FROM excluded_positions);
+	`
+
+	var firstID int
+	var totalCount int
+	if err := db.db.QueryRow(statsQuery).Scan(&firstID, &totalCount); err != nil {
+		return nil, fmt.Errorf("error getting position stats: %w", err)
+	}
+
+	// Calculate the interval size
+	interval := float64(totalCount) / float64(count)
+	intervalInt := int(math.Round(interval))
+
+	// Now get the evenly distributed positions
+	const positionsQuery = /* sql */ `
+		WITH excluded_positions AS (
+			SELECT id FROM positions ORDER BY id DESC LIMIT 100
+		)
```

Comment:
We could maybe combine all this into one statement so we can reused the excluded positions etc.